build/bin/remove_unneeded

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+
+# Sanity check: the file is created by the Dockerfile.
+if [ ! -f /in-the-container ]; then
+  echo "Don't run this outside the container!"
+  exit 1
+fi
+
+# Array of basic tools needed by entrypoint and user_setup.
+basicNeed=("bin/sh" "/bin/bash" "/usr/bin/coreutils")
+libraries=()
+
+# Get all that is needed by basicNeed
+for i in "${basicNeed[@]}"; do
+    ldd=$(ldd "${i}" \
+        | grep so \
+        | sed -e '/^[^\t]/ d' \
```

Comment:
Is it possible to simplify all these `sed`, `sort`, `awk` etc commands? For example, In my limited testing, the following pulls out the shared library path:

```
ldd /usr/bin/bash | grep -P "\.so\.\d?" | awk '{print $1}'
```

Reply:
I simplified a little bit but if it's OK I would prefer keeping the sed as it gives me full path instead of just library name.